### PR TITLE
Feature/59 submit tip

### LINKED ISSUE #59: Polkadot treasury motion #29 | Milestone-1 intake review | Unable to propose the `Tip`
Unable to propose the `Tip`

After choosing "OK" no response

**Validation Setup**

| Component | Version |
| --- | --- |
| Release APK | litentry-governance_v1.0.1.apk |
| Android Version | 7.1.1 |
| Litentry Devnet | wss://18.140.130.138 |
| Parity Signer | v4.5.3-beta |

### PR DESCRIPTION
**Pull request by ahoseinian**
Simple page that runs `tip.reportAwesome` api. 

the transaction is not completing in my simulator maybe @geastwood can provide a way to test the signing process? [resolved]

https://user-images.githubusercontent.com/1168983/121381067-ae5a1e00-c945-11eb-9381-79dabe866087.mp4

> Note: this PR also tested with signing process.

### CODE REVIEW

**Comment by geastwood** [comment]
Let's make this PR `WIP`, and have a session to show and document how to use `Parity Signer` to construct a valid transaction.

**Comment by geastwood** [comment]
Is this PR ready to be reviewed again?

**Comment by ahoseinian** [comment]
@geastwood first review the https://github.com/litentry/litentry-app/pull/123 please

File: src/context/TxContext/index.tsx [hunk 1]
```diff
@@ -50,11 +50,15 @@ function TxContextProvider({children}: PropTypes): React.ReactElement {
     const {api, txMethod, params, description, address, title} = config;
     const [section, method] = txMethod.split('.');
 
+    if (!section || !method) {
+      throw new Error('section or method undefined');
```
Comment (geastwood) [inline src/context/TxContext/index.tsx:54]: How is this error being handled?

File: src/screen/MyIdentityScreen/DisplayJudgement.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ function DisplayJudgement(props: PropTypes) {
   const {api} = useContext(ChainApiContext);
   const successMsg = `This address has ${judgementCount} judgement${
     judgementCount > 1 ? 's' : ''
-  } from Registrar ${detail.data?.judgements.map((judgement) => `#${judgement[0]}`).join(',')}. It's all set. 🎉`;
+  } from Registrar ${detail.data?.judgements?.map((judgement) => `#${judgement[0]}`).join(',')}. It's all set. 🎉`;
```
Comment (geastwood) [inline src/screen/MyIdentityScreen/DisplayJudgement.tsx]: how come judgements is nullable now? If that is the case the string message won't make sense no?

File: src/screen/SubmitTipScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        renderTitle={() => (
+          <Text category={'s1'} style={globalStyles.monoFont}>
+            Submit Tip
+          </Text>
+        )}
+        accessoryLeft={
+          <TopNavigationAction onPress={navigation.goBack} icon={(p) => <Icon {...p} name={'arrow-back-outline'} />} />
+        }
+      />
+      <SafeAreaView edges={['bottom']} style={styles.container}>
```
Comment (geastwood) [inline src/screen/SubmitTipScreen.tsx:53]: Shouldn't this `SafeAreaView` be the most outside wrapper?

**Review by geastwood — DISMISSED** [review]

File: src/screen/MyIdentityScreen/DisplayJudgement.tsx (hunk 2 above)
Comment (ahoseinian) [reply src/screen/MyIdentityScreen/DisplayJudgement.tsx]: it's not, don't know why I added that `?`.  it's now like before.

File: src/context/TxContext/index.tsx (hunk 1 above)
Comment (ahoseinian) [reply src/context/TxContext/index.tsx:54]: it'll be caught by `start(...args).catch()` so no crashes. the check is basically identical to the next if check
```
if (!get(api.tx, [section, method])) {
```
TS needs this to understand `section` & `method` are not `undefined`.

File: src/screen/SubmitTipScreen.tsx (hunk 3 above)
Comment (ahoseinian) [reply src/screen/SubmitTipScreen.tsx:53]: `ScreenNavigation.tsx` already has a safeAreaView in it, so we can't wrap that with another `SafeAreaView`. I can refactor that in another PR to have only one SafeAreaView per page. what do you think?

Comment (geastwood) [reply src/screen/SubmitTipScreen.tsx:53]: It sounds good.

**Review by geastwood — APPROVED** [review]
Good work!

**Pull request merged**